src/test/java/org/reactome/addlinks/test/referencecreators/TestEnsemblBioMartMicroarrayPopulator.java

```diff
@@ -0,0 +1,103 @@
+package org.reactome.addlinks.test.referencecreators;
+
+import org.gk.model.GKInstance;
+import org.gk.model.ReactomeJavaConstants;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.reactome.addlinks.referencecreators.EnsemblBioMartMicroarrayPopulator;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.*;
+import static org.junit.Assert.*;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@ContextConfiguration("/test-application-context.xml")
+@RunWith(org.springframework.test.context.junit4.SpringJUnit4ClassRunner.class)
+
+public class TestEnsemblBioMartMicroarrayPopulator {
+
+    @Autowired
+    EnsemblBioMartMicroarrayPopulator microarrayPopulator;
+
+    @Mock
+    GKInstance mockRGPInst;
+
+    @Mock
+    GKInstance mockRGPInst2;
+
+    @Mock
+    Map<String, List<String>> mockMap;
+
+    @Mock
+    GKInstance mockRefDBInst;
+
+    Map<String, Map<String, List<String>>> mappings = new HashMap<>();
+
+    @Before
+    public void setup() {
+        MockitoAnnotations.initMocks(this);
+
+        Map<String, List<String>> mockProteinToTranscripts = new HashMap<>();
+        mockProteinToTranscripts.put("ENSP12345", Arrays.asList("ENST12345"));
+        Map<String, List<String>> mockTranscriptToMicroarrays = new HashMap<>();
+        mockTranscriptToMicroarrays.put("ENST12345", Arrays.asList("M12345"));
+        Map<String, List<String>> mockUniprotToProteins = new HashMap<>();
+        mockUniprotToProteins.put("A7MBE8", Arrays.asList("ENSP12345"));
+        mappings.put("btaurus_proteinToTranscripts", mockProteinToTranscripts);
+        mappings.put("btaurus_transcriptToMicroarrays", mockTranscriptToMicroarrays);
+        mappings.put("btaurus_uniprotToProteins", mockUniprotToProteins);
+    }
+
+    @Test
+    public void testEnsemblBiomartMicroarrayPopulator() {
+
+        List<GKInstance> mockSourceReferences = Arrays.asList(mockRGPInst);
+        System.setProperty("config.location", "src/test/resources/addlinksTest-btau.properties");
+        try {
+            microarrayPopulator.setTestMode(true);
+            microarrayPopulator.createIdentifiers(12345L, mappings, mockSourceReferences);
+        } catch (Exception e) {
+            e.printStackTrace();
+            fail();
+        }
```

Comment:
I think here it should be possible to add `throws Exception` to the method signature in place of the try/catch.

Reply:
Wouldn't that mask the testing goal? If I had the method throw an Exception, it would just be calling `createIdentifiers` without there being anything to assert

Reply:
Fair point - to clarify, is this test expecting an exception to be thrown?  If so, to be more explicit about the expected exception you could use this method of testing:
https://github.com/reactome/add-links/pull/134#discussion_r375549443 or add assert statements in the catch block.

Reply:
No, this is just meant to confirm the code ran correctly. I was unsure how to test it due to the lack of returned data, and so I followed Solomon's example: https://github.com/reactome/add-links/blob/develop/src/test/java/org/reactome/addlinks/test/referencecreators/TestKEGGReferenceCreator.java

Reply:
Ah okay, I see.  Is there a return value or data structure you could test to see if something is empty?  If not, this is okay but if you want you could use the verify method from Mockito to ensure whatever you are expecting to run is actually run: 

https://stackoverflow.com/questions/9841623/mockito-how-to-verify-method-was-called-on-an-object-created-within-a-method

Reply:
The only feedback the ReferenceCreators/Populators have is the fact they haven't failed, since its purely a DB writing step. I think for now I am OK with this way of doing it

Reply:
Okay, sounds good